framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeCopier.java

```diff
@@ -293,6 +293,10 @@ public AnnotatedTypeMirror visitTypeVariable(
     public AnnotatedTypeMirror visitPrimitive(
             AnnotatedPrimitiveType original,
             IdentityHashMap<AnnotatedTypeMirror, AnnotatedTypeMirror> originalToCopy) {
+        if (originalToCopy.containsKey(original)) {
+            return originalToCopy.get(original);
+        }
```

Comment:
Does this change solve your problem?
Because I don't see what effect this change has... shouldn't the returned value be added to `originalToCopy` before returning?
Can you add some comment what this is doing?

Reply:
Yep this changes solve the problem.